### ISSUE TITLE
SC2208 is too permissive

### ISSUE THREAD

**Issue by mbs-c** [koalaman/shellcheck#2755]
#### For bugs
- Rule Id: [SC2208](https://www.shellcheck.net/wiki/SC2208)
- My shellcheck version: online
- [x] The rule's wiki page does not already cover this
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

key='$id'
readonly -A array=( ['$id']='my_id' )
[[ -v "array[$key]" ]]; echo $?
```

#### Here's what shellcheck currently says:

Nothing about line 5, which is where the actual bug is.
(SC2016 is emitted twice, but only because I used single-quoted $ to demonstrate the problem.)

#### Here's what I wanted or expected to see:

```text
[[ -v "array[$key]" ]]; echo $?
       ^-- SC2208 (error): Single-quote array arguments to -v to avoid confusing expansions.
```

(The suggested message is different from the current description of SC2208.)

#### Background

SC2208 is too permissive, since it does not take associative arrays into account:
 * it only applies to single-bracket test expressions
 * the wiki incorrectly suggests that using double brackets will avoid all issues with '-v' and arrays, which is not the case in both bash and zsh (and potentially other shells?)
 * it does not complain about a double-quoted arguments referring to array keys, even though they are also potentially unsafe (see my example)

Both the wiki page for SC2208 and the check itself should be updated to give more accurate advice.
